src/utils/objectUtils.ts

```diff
@@ -56,29 +56,47 @@ export function pickWithoutUndefined<T, K extends string | number | symbol>(
 
 export function isEmptyObject(params: Record<string, unknown>): boolean {
   for (const key in params) {
-    if (Object.prototype.hasOwnProperty.call(params, key) && params[key] !== undefined) {
+    if (Object.hasOwn(params, key) && params[key] !== undefined) {
       return false
     }
   }
   return true
 }
 
-export function groupBy<T>(inputArray: T[], propName: string): Record<string, T[]> {
-  return inputArray.reduce((result, entry) => {
-    // @ts-ignore
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    const key = entry[propName]
+type RecordKeyType = string | number | symbol
+export function groupBy<T extends { [K in keyof T]: unknown }, K extends keyof T>(
+  array: T[],
+  selector: K,
+): Record<RecordKeyType, T[]> {
+  return array.reduce(
+    (acc, item) => {
+      if (!item) {
+        return acc
+      }
+      const key = item[selector] as RecordKeyType
+      if (!acc[key]) {
+        acc[key] = []
+      }
+      acc[key].push(item)
+      return acc
+    },
+    {} as Record<RecordKeyType, T[]>,
+  )
+}
 
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-    if (Object.hasOwnProperty.call(result, key)) {
-      // @ts-ignore
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,@typescript-eslint/no-unsafe-call
-      result[key].push(entry)
-    } else {
-      // @ts-ignore
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-      result[key] = [entry]
-    }
-    return result
-  }, {})
+export function groupByUnique<T extends { [K in keyof T]: unknown }, K extends keyof T>(
```

Comment:
@kibertoad I am still not sure if it is better to have a different method or include a boolean property, but if you think this is a better solution let's go with it